src/main/java/org/spongepowered/api/item/data/MappedItemData.java

```diff
@@ -0,0 +1,143 @@
+ * @param <V> The value type
+ * @param <T> The intended comparable extended type
+ */
+public interface MappedItemData<K, V, T extends MappedItemData<K, V, T>> extends ItemData<T> {
+
+    /**
+     * Gets all the mapped key values stored on the item stack.
+     *
+     * <p>The returned set is an immutable one, only representing the
+     * mapped key values available on this {@link MappedItemData}. To modify
+     * the available mapped key values, use the provided methods in
+     * {@link MappedItemData}.</p>
+     *
+     * @return All mapped key values stored to the item stack
+     */
+    Set<K> getKeys();
+
+    /**
+     * Gets all the mapped key values and the levels stored on the item stack.
+     *
+     * <p>The returned map is an immutable one, only representing the
+     * mapped key values available on this {@link MappedItemData}. To modify
+     * the available mapped key values, use the provided methods in
+     * {@link MappedItemData}.</p>
+     *
+     * @return An immutable map of all keys and their values
+     */
+    Map<K, V> asMap();
+
+    /**
+     * Gets the value of the desired key on the
+     * item stack.
+     *
+     * @param key The key
+     * @return The value of the key, if available
+     */
+    Optional<V> get(K key);
+
+    /**
+     * Sets the key with the desired value.
+     *
+     * <p>This will perform any and all checks that the given key
+     * is compatible with the represented item, and that the value is within
+     * the bounds of key.</p>
+     *
+     * @param key The key to set
+     * @param value The value of the key
+     * @return The transaction result, if the keys and their respective values
+     *     were added, rejected or replaced
+     */
+    ItemDataTransactionResult set(K key, V value);
+
+    /**
+     * Sets mapped data attached to the item stack.
+     *
+     * <p>This will perform any and all checks that the given key
+     * is compatible with the represented item, and that the value is within
+     * the bounds of key.</p>
+     *
+     * @param mapped The mapped data attached to the item stack
+     * @return The transaction result, if the keys and their respective values
+     *     were added, rejected or replaced
+     */
+    ItemDataTransactionResult set(Map<K, V> mapped);
+
+    /**
+     * Sets mapped data attached to the item stack.
+     *
+     * <p>This will set all the mapped with their default value.</p>
+     *
+     * @param mapped The mapped data attached to the item stack
+     * @return The transaction result, if the mapped were added rejected or replaced
+     */
+    ItemDataTransactionResult set(K... mapped);
+
+    /**
+     * Unsafely sets the key at the desired value.
+     *
+     * <p>This does NOT perform any checks on the item or whether
+     * the value is within the standard bounds of key.</p>
+     *
+     * @param key The key to add
+     * @param value The value of the key to add
+     */
+    void setUnsafe(K key, V value);
```

Comment:
That's not the point of this method.... It's like setting unsafe enchantment levels on an item, you DON'T want to throw exceptions when the values are outside of their norm.